### ISSUE TITLE
Mocking.itThrows: allow checked exceptions

### ISSUE THREAD

**Issue by thetric** [MarkusAmshove/Kluent#109]
Mocks should be allowed to throw checked exceptions. Currently only ``RuntimeException``s and ``Error``s are allowed. The following test doesn't compile:

```kt
object MockingSpec : Spek({
    describe("Mocking.kt") {
        it("mocks should be allow to throw checked exceptions") {
            val sut: Callable<Any> = mock()
            When calling sut.call() itThrows IOException("")
        }
    }
})
```

Mockito allows to specify any `Throwable` in its `thenThrow` clause: [JavaDoc](https://static.javadoc.io/org.mockito/mockito-core/2.18.3/org/mockito/stubbing/OngoingStubbing.html#thenThrow-java.lang.Class-)

It would be sufficient to change the parameter type of the following files' first function from `RuntimeException` to `Throwable` and delete the second function with the `Error` parameter:

https://github.com/MarkusAmshove/Kluent/blob/c970c54ecaa15895186dfd65f7e8022752cbbb5f/src/main/kotlin/org/amshove/kluent/Mocking.kt#L44

https://github.com/MarkusAmshove/Kluent/blob/c970c54ecaa15895186dfd65f7e8022752cbbb5f/src/main/kotlin/org/amshove/kluent/Mocking.kt#L46

---

https://github.com/MarkusAmshove/Kluent/blob/c970c54ecaa15895186dfd65f7e8022752cbbb5f/src/main/kotlin/org/amshove/kluent/MockingBacktick.kt#L8

https://github.com/MarkusAmshove/Kluent/blob/c970c54ecaa15895186dfd65f7e8022752cbbb5f/src/main/kotlin/org/amshove/kluent/MockingBacktick.kt#L9

**Comment by MarkusAmshove** [comment]
Thank you for reporting this!

I've changed it within PR #106 which will be merged sometime this week.

Note that in the future (next major release) mocking will be deprecated (#73).
There might be the possibility to add a library which depends on Kluent and keeps the mocking features if there is enough demand, but that isn't decided yet.

There will be a 1.x version which doesn't have any new features but just includes all the deprecation annotations to see if your codebase would be compatible with 2.x and also provide automatic conversions where possible.

**Comment by MarkusAmshove** [comment]
This is within the just released 1.39
